aptos-move/move-examples/mint_nft/step3/sources/nft_step3.move

```diff
@@ -0,0 +1,240 @@
+
+        // Create the nft collection.
+        token::create_collection(resource_signer, collection_name, description, collection_uri, maximum_supply, mutate_setting);
+
+        // Create a token data id to specify the token to be minted.
+        let token_data_id = token::create_tokendata(
+            resource_signer,
+            collection_name,
+            token_name,
+            string::utf8(b""),
+            0,
+            token_uri,
+            signer::address_of(resource_signer),
+            1,
+            0,
+            // This variable sets if we want to allow mutation for token maximum, uri, royalty, description, and properties.
+            // Here we enable mutation for properties by setting the last boolean in the vector to true.
+            token::create_token_mutability_config(
+                &vector<bool>[ false, false, false, false, true ]
+            ),
+            // We can use property maps to record attributes related to the token.
+            // In this example, we are using it to record the receiver's address.
+            // We will mutate this field to record the user's address
+            // when a user successfully mints a token in the `mint_nft()` function.
+            vector<String>[string::utf8(b"given_to")],
+            vector<vector<u8>>[b""],
+            vector<String>[ string::utf8(b"address") ],
+        );
+
+        // store the token data id within the module, so we can refer to it later
+        // when we're minting the NFT
+        let resource_signer_cap = resource_account::retrieve_resource_account_cap(resource_signer, @source_addr);
+        move_to(resource_signer, ModuleData {
+            signer_cap: resource_signer_cap,
+            token_data_id,
+            minting_enabled: false,
+            expiration_timestamp: 10000000000,
+        });
+    }
+
+    /// Mint an NFT to the receiver. Note that different from the tutorial in step1, here we only ask for the receiver's
+    /// signer. This is because we used resource account to publish this module and stored the resource account's signer
+    /// within the `ModuleData`, so we can programmatically sign for transactions instead of manually signing transactions.
+    /// See https://aptos.dev/concepts/accounts/#resource-accounts for more details.
+    public entry fun mint_nft(receiver: &signer) acquires ModuleData {
+        // Mint token to the receiver.
+        let module_data = borrow_global_mut<ModuleData>(@mint_nft);
+
+        // Check the config of this module to see if we enable minting tokens from this collection
+        assert!(timestamp::now_seconds() < module_data.expiration_timestamp, error::permission_denied(ECOLLECTION_EXPIRED));
+        assert!(module_data.minting_enabled, error::permission_denied(EMINTING_DISABLED));
+
+        let resource_signer = account::create_signer_with_capability(&module_data.signer_cap);
+        let token_id = token::mint_token(&resource_signer, module_data.token_data_id, 1);
```

Comment:
Not sure if this increase the scope of PR. happy to see a followup PR as well. 
I think we have call this `mint_event_ticket` instead of mint_nft. when people see mint nft, they will directly think of mint pfp nft. Here we are actually creating NFT print. 
An interesting example is be using this contract for minting event pass. Each nft is an event ticket, which has two properties: (1) expiration_sec and (2) ticket_used. 

when we mint, we can set an expiration sec for the NFT and ticket_used as false
we can also have a function to modify pass_used field  to true if the user used the ticket. 

The code would most stay the same here. The change would be mostly adding a story/application around this contract.

Reply:
Good point! Yeah I feel like a lot of people are confused on this.
I added some comments [here](https://github.com/aptos-labs/aptos-core/blob/156d6376f51656e31a6c85d187b151a2bc806aea/aptos-move/move-examples/mint_nft/1-Create-NFT/sources/create_nft.move#L4) to address that this tutorial is for creating printing editions of a base token. 

In the future, we can add a new tutorial that illustrates how to mint an pfp NFT.

Reply:
sounds good. This is a lot better than status quo.